bin/describe/describe_pp.ml

```diff
@@ -29,6 +19,28 @@ let pp_with_ocamlc env ~ocamlc dialects pp_file =
     User_error.raise [ Pp.textf "cannot find a dump file: %s" (Path.to_string dump_file) ]
 ;;
 
+let files_for_source file dialects =
+  let base, ext = Path.split_extension file in
+  let dialect, kind =
+    match Dune_rules.Dialect.DB.find_by_extension dialects ext with
+    | None -> User_error.raise [ Pp.textf "unsupported extension: %s" ext ]
+    | Some x -> x
+  in
+  let new_ext =
+    ext ^ Option.value (Dune_rules.Dialect.ml_suffix dialect kind) ~default:""
```

Comment:
I think for readability `sprintf "%s%s"` would be nicer. Or using `Path.extend_basename` which does exactly this under the hood, e.g. `Path.extend_basename file ~suffix:(Option.value ..)`.